api/src/main/java/org/apache/iceberg/catalog/Catalog.java

```diff
@@ -120,6 +120,14 @@ default boolean tableExists(TableIdentifier identifier) {
    */
   boolean dropTable(TableIdentifier identifier);
 
+  /**
+   * Drop a table and remove all data and metadata files.
+   *
+   * @param identifier a table identifier
+   * @return true if the table was dropped, false if the table did not exist
+   */
+  boolean dropTableAndData(TableIdentifier identifier);
```

Comment:
Is another call API method necessary (it has the same parameter) or could the behavior be represented by a boolean parameter in `dropTable`? I think `dropTableAndData` adds the assumption that `dropTable` does not delete the data but that is not specified in the `dropTable` docs either so it makes it confusing.

I think avoiding another method call could help keeping the API simple.

Reply:
I think it is better to add a second call than to add a boolean parameter, since Java doesn't support calling methods using names. It isn't clear how `dropTable(name, true)` differs from `dropTable(name, false)`.

Reply:
> It isn't clear how dropTable(name, true) differs from dropTable(name, false)

This is not specific to `boolean` parameters, but any Java argument. However, usually this is clarified with clear javadocs and I agree that it would not be a good choice if multiple `boolean` parameters were in the method signature - however, this is not the case here.

Still, having multiple drop methods seems confusing if there's not a clear idea of what each of them do and how do they differ, which is still a problem with `dropTable`, since the JavaDoc does not specify how it differs from `dropTableAndData`.

Reply:
I agree, either way the docs should be updated.

Reply:
Okay, I've updated this. The new behavior is to drop all table data in `dropTable`, which is documented in javadoc. I've also added a variant of `dropTable` with a `purge` flag that specifies whether to delete data files or not.

Reply:
Hey, @rdblue taking a look, I'm not sure I see the updates that you mentioned.  I assume what you meant to say was that `dropTable` drops all table metadata, but not data files.  I don't see that in the javadoc.

Reply:
@danielcweeks, the new purge option will delete all data and metadata files, not just metadata files. The default `dropTable` calls `dropTable` with purge enabled. There is also a variant where you can set purge to `false` to avoid deleting data.